### ISSUE TITLE
train data proportion

### ISSUE THREAD

**Issue by wuzhiyang2016** [foreverYoungGitHub/MTCNN#20]
Hi  @foreverYoungGitHub  , in your reply :  For example, the positive : part : negative= 1 : 3 : 3 at the beginning, while it will change to positive : part : negative= 1 : 5 : 3 in the next iteration.
     you mean in a train process, we need to change every batchsize‘s data proportion ?  or you mean we use 1:3:3 to train a model  A.caffemodel ,then,we use this model to generate train data, the proportion set to 1:5:3 ,finetune on A.caffemodel  ?  then we will get B.caffemodel

**Comment by wuzhiyang2016** [comment]
and we set part-face label is 1, positive-face label also 1,   i mean these two's number is more than negative-samples,,,it is ok?  because i train an 1:3:3，learning rate is 0.0001,,, model performace is bad, and the train loss is always keep to 0.5,,,maybe there are something wrong~ @foreverYoungGitHub

**Comment by foreverYoungGitHub** [comment]
Yes, you are right!

The proportion is not suitable. You need to keep the negative face : (positive face and part face) around 3 : 1. But the proportion of positive face and part face could be adjusted based on the iteration.

It would better to write a script to automatically sample the train set and train the face.

In addition, it would be better to use hard data mining to find the hard data in each iteration and retrain the net based on the hard data and sampled previous data.
